network/wsNetwork.go

```diff
@@ -2027,6 +2122,7 @@ func (wn *WebsocketNetwork) removePeer(peer *wsPeer, reason disconnectReason) {
 		if peer.throttledOutgoingConnection {
 			atomic.AddInt32(&wn.throttledOutgoingConnections, int32(1))
 		}
+		atomic.AddInt32(&wn.peersChangeCounter, 1)
```

Comment:
I understand the idea of `peersChangeCounter` but not sure I like adding even more state to not the simplest code.
I'd suggest encapsulation but peers array is already heavily used around.

Reply:
Yes, it's not the simplest code, and I agree that we should see if it can be encapsulated somehow without negative performance impact.